lib/csv/parser.rb

```diff
@@ -335,20 +364,26 @@ def prepare_regexp
       @column_separator = @options[:column_separator].to_s.encode(@encoding)
       @row_separator =
         resolve_row_separator(@options[:row_separator]).encode(@encoding)
-      @quote_character = @options[:quote_character].to_s.encode(@encoding)
-      if @quote_character.length != 1
-        raise ArgumentError, ":quote_char has to be a single character String"
-      end
+
+
       @backslash_character = "\\".encode(@encoding)
+      if @options[:quote_character].nil?
+        @quote_character = nil
+      else
+        @quote_character = @options[:quote_character].to_s.encode(@encoding)
+        if @quote_character.length != 1
+          raise ArgumentError, ":quote_char has to be a single character String"
+        end
+        escaped_quote_character = Regexp.escape(@quote_character)
+        @escaped_quote = Regexp.new(escaped_quote_character)
+        @backslash_quote_character = @backslash_character + escaped_quote_character
+      end
 
       escaped_column_separator = Regexp.escape(@column_separator)
       escaped_first_column_separator = Regexp.escape(@column_separator[0])
       escaped_row_separator = Regexp.escape(@row_separator)
-      escaped_quote_character = Regexp.escape(@quote_character)
       escaped_backslash_character = Regexp.escape(@backslash_character)
       @escaped_backslash = Regexp.new(escaped_backslash_character)
-      @escaped_quote = Regexp.new(escaped_quote_character)
-      @backslash_quote_character = @backslash_character + escaped_quote_character
```

Comment:
Could you initialize them here to collect `escaped_*` related variables the same code block?

```ruby
if @quote_character
  escaped_quote_character = Regex.escape(@quote_character)
  @escaped_quote = Regexp.new(escaped_quote_character)
  @backslash_quote_character = @backslash_character + escaped_quote_character
end
```

Reply:
Fixed it. https://github.com/ruby/csv/pull/79/commits/b5233519e4bc00d3ff9d605c9bacb3ad5dcfe43b